src/services/refactors/convertArrowFunctionOrFunctionExpression.ts

```diff
@@ -178,7 +178,10 @@ namespace ts.refactor.convertArrowFunctionOrFunctionExpression {
 
         const { variableDeclaration, variableDeclarationList, statement, name } = variableInfo;
         suppressLeadingTrivia(statement);
-        const newNode = factory.createFunctionDeclaration(func.decorators, statement.modifiers, func.asteriskToken, name, func.typeParameters, func.parameters, func.type, body);
+
+        const modifiersFlags = getCombinedModifierFlags(variableDeclaration) | getEffectiveModifierFlags(func);
+        const modifiers = factory.createModifiersFromModifierFlags(modifiersFlags);
+        const newNode = factory.createFunctionDeclaration(func.decorators, length(modifiers) ? modifiers : undefined, func.asteriskToken, name, func.typeParameters, func.parameters, func.type, body);

```

Comment:
Why not just pass in `modifiers`?

```suggestion
        const newNode = factory.createFunctionDeclaration(func.decorators, modifiers, func.asteriskToken, name, func.typeParameters, func.parameters, func.type, body);
```

Reply:
Actually if i pass `modifiers` in, the `textChanges` does not happy and break an assert.

Reply:
Out of curiosity, which assert? We probably should consider converting an empty modifier list into `undefined` in the factory, since our parser never produces an empty modifier list (if it fails to parse modifiers, it returns `undefined`).

Reply:
```
Error: Debug Failure. False expression.
      at getPos (src\services\textChanges.ts:10:15)
      at assignPositionsToNodeArray (src\services\textChanges.ts:1030:39)
      at Object.visitEachChild (src\compiler\visitorPublic.ts:876:21)
      at assignPositionsToNode (src\services\textChanges.ts:1016:25)
      at getNonformattedText (src\services\textChanges.ts:999:52)
      at getFormattedTextOfNode (src\services\textChanges.ts:973:36)
      at format (src\services\textChanges.ts:951:41)
      at computeNewText (src\services\textChanges.ts:954:19)
```